finesse/gui/hardware_set/device_view.py

```diff
@@ -319,7 +319,7 @@ def _on_open_close_clicked(self) -> None:
 class DeviceControl(QGroupBox):
     """Allows for viewing and connecting to devices."""
 
-    def __init__(self, connected_devices: AbstractSet[OpenDeviceArgs]) -> None:
+    def __init__(self, connected_devices: Set[OpenDeviceArgs]) -> None:
```

Comment:
Out of curiosity, is this something you have decided to fix here simply because of convenience, or has it been a sort of recommendation/fix done by ruff? Either way, why the change?

Reply:
It was a recommendation from ruff. Apparently using `typing.AbstractSet` is deprecated and we should be using `collections.abc.Set` instead.